Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm

```diff
@@ -616,9 +616,19 @@ static NSHTTPCookieAcceptPolicy httpCookieAcceptPolicy(CFHTTPCookieStorageRef co
 
 void NetworkStorageSession::deleteCookies(const ClientOrigin& origin, CompletionHandler<void()>&& completionHandler)
 {
-    auto cachePartition = origin.topOrigin == origin.clientOrigin ? emptyString() : ResourceRequest::partitionName(origin.topOrigin.host());
-    deleteCookiesMatching([domain = origin.clientOrigin.host(), &cachePartition](NSHTTPCookie* cookie) {
-        return String(cookie.domain) == domain && equalIgnoringNullity(String(cookie._storagePartition), cachePartition);
+    auto cachePartition = cookiePartitionIdentifier(origin.topOrigin);
+    bool isMainFrame = origin.topOrigin == origin.clientOrigin;
+    deleteCookiesMatching([domain = origin.clientOrigin.host(), &cachePartition, isMainFrame](NSHTTPCookie* cookie) {
+        if (isMainFrame) {
+            if (String(cookie.domain) == domain && !cookie._storagePartition)
+                return true;
+            if (!cachePartition.isNull() && equalIgnoringNullity(String(cookie._storagePartition), cachePartition))
```

Comment:
Does this mean a first-party response can clear data for cookies set by a third-party response?
Is this specified in spec?  I don't think this change is consistent with other storage types.
For the other types, the delete operation with `ClientOrigin` (e.g. {OriginA, OriginA}) will only clear data in that partition (e.g. OriginA/OriginA, not OriginA/OriginB).

Reply:
Unless I'm misinterpreting the proposed spec change, yes the first party can clear third-party partitioned cookies: https://datatracker.ietf.org/doc/html/draft-cutler-httpbis-partitioned-cookies#name-partitioned-cookies-and-cle

Reply:
Per discussion offline, we should limit the deletion to its own partition.